docker/build_artifacts/sagemaker/serve.py

```diff
@@ -67,6 +67,16 @@ def __init__(self):
         self._gunicorn_timeout_seconds = int(
             os.environ.get("SAGEMAKER_GUNICORN_TIMEOUT_SECONDS", 30)
         )
+        self._nginx_proxy_read_timeout_seconds = int(
+            os.environ.get("SAGEMAKER_NGINX_PROXY_READ_TIMEOUT_SECONDS", 60))
+
+        # Nginx proxy read timeout should not be less than the GUnicorn timeout. If it is, this
+        # can result in upstream timed out errors.
+        if self._gunicorn_timeout_seconds > self._nginx_proxy_read_timeout_seconds:
+            self._nginx_proxy_read_timeout_seconds = self._gunicorn_timeout_seconds
```

Comment:
nit: Can we also log the `nginx_proxy_read_timeout` values before and after as that is getting modified here?